### PR TITLE
Feature/encoder decoder upgrade

### PR DESCRIPTION
**Pull request by azadeh-gh**
### Summary
This PR introduces several improvements and optimizations to the GNN model and evaluation pipeline to enhance prediction quality and address CUDA out-of-memory (OOM) issues encountered during 1-month global ATMS runs.
### Key Changes

**1. Model Architecture Enhancements**

- Increased the number of encoder and hidden layers.
- Expanded hidden dimension size.
- Replaced MSE loss with Huber loss, and added weighted Huber loss for more robust and channel-aware supervision.

**2. Training Configuration**
- Implemented `configure_optimizers()` in PyTorch Lightning using the Adam optimizer and a learning rate scheduler for more controlled training.

**3. Decoder Optimization**

- Introduced chunked decoding to prevent CUDA OOM issues by processing large decoder input tensors in smaller, manageable slices.

**4. Evaluation & Visualization Improvements**

- Updated evaluation scripts to produce:
  - Map plots for ATMS and surface pressure predictions.
  - Error percentage plots for quick diagnostics.
### Benchmark Results

The following benchmark outputs are attached for model evaluation:

 **1-month run ATMS Predictions**

bt_map_channel_1

hist_norm_ytrue_ypred_ch_1_epoch10

 **2-week run ATMS  and Surface Pressure Predictions**

bt_map_channel_1 (4)
hist_bt_ch_1_bt_only_epoch9
pressure_map (3)
hist_pressure_epoch9

**These plots serve as a baseline to evaluate scan pattern artifacts and prediction quality for future changes.**

Known Issues & Follow-ups
[#35 ] Predicted pressure shows input scan pattern
The predicted pressure fields exhibit satellite scanning artifacts similar to the input scan pattern. This may suggest the model is overfitting to input scan geometry.
Action: Run the model on longer time periods and evaluate if the issue persists.

[#35 ] Predicted scan angles mirror input scan angles
The predicted scan angle appears to follow the input rather than the target. This could be due to incorrect supervision setup or improper handling of scan angle features.
Action: Ensure the target scan angle is correctly set and used during decoding and loss computation

[#36 ] Missing values in output plots
Some plots contain unexplained missing values (NaNs or gaps).
Action: Investigate whether this is due to missing targets, masking logic, or plotting artifacts.

### CODE REVIEW

File: gnn_model/gnn_model.py [hunk 1]
```diff
@@ -318,9 +329,27 @@ def forward(self, data):
...
+            for start in range(0, num_rows, chunk_size):
+                end = min(start + chunk_size, num_rows)
+                chunk = decoder_input[start:end]  # stays on GPU
+                decoded_chunks.append(self.decoder_mlp(chunk))
+
+            decoded = torch.cat(decoded_chunks, dim=0)
+        else:
+            decoded = self.decoder_mlp(decoder_input)
+
+        out = decoded
```
[the remark this answers is not in the document]
Comment (xincjin-NOAA) [reply gnn_model/gnn_model.py:347]: self.decoder_mlp(chunk) will use less GPU memory

[the remark this answers is not in the document]
Comment (rmclaren) [reply gnn_model/gnn_model.py:347]: @azadeh-gh Thanks for the clarification!
